utils/tangleid/BaseExtension.js

```diff
@@ -0,0 +1,31 @@
+
```

Comment:
Why is there leading space line?

Reply:
It is the rule that recommends by the style guide and ESLint explains why we need it. [eol-last](https://eslint.org/docs/rules/eol-last)

> Trailing newlines in non-empty files are a common UNIX idiom. Benefits of trailing newlines include the ability to concatenate or append to files as well as output files to the terminal without interfering with shell prompts.